File: stringio/CutStrL.c
// This file is part of Adlib Tracker II (AT2).
//
// SPDX-FileType: SOURCE
// SPDX-FileCopyrightText: 2014-2024 The Adlib Tracker 2 Authors
// SPDX-License-Identifier: GPL-3.0-or-later

String_t CutStrL (const String *str, uint8_t margin) {
  String_t r;
  size_t len = Length (str), i = 0;
  const unsigned char *s = GetStr (str);

  if ((!margin) || (margin > len))
    margin = len;

  while ((i < margin) && (s[i] == ' '))
    i++;

  len -= i;
  r.len = len;
  if (len)
    memcpy (r.str, s + i, len);

  return r;
}

File: stringio/pas/CutStrL.pas
// This file is part of Adlib Tracker II (AT2).
//
// SPDX-FileType: SOURCE
// SPDX-FileCopyrightText: 2014-2024 The Adlib Tracker 2 Authors
// SPDX-License-Identifier: GPL-3.0-or-later

function CutStrL(str: String; margin: Byte): String;

var
  idx: Byte;

begin
  If (margin = 0) then margin := Length(str)
  else If (margin > Length(str)) then
         margin := Length(str);
  idx := 0;
  While (idx+1 <= margin) and (str[idx+1] = ' ') do
    Inc(idx);
  If (idx <> 0) then Delete(str,1,idx);
  CutStrL := str;
end;
